lib/src/global.dart

```diff
@@ -17,3 +17,87 @@ Future<void> installOfflineMapTiles(String tilesDb) async {
     },
   );
 }
+
+Future<List<OfflineRegion>> getListOfRegions({String accessToken}) async {
+  String regionsJson = await _globalChannel.invokeMethod(
+    'getListOfRegions',
+    <String, dynamic>{
+      'accessToken': accessToken,
+    },
+  );
+  Iterable regions = json.decode(regionsJson);
+  return regions.map((region) => OfflineRegion.fromJson(region)).toList();
```

Comment:
As as suggestion, we could follow this article https://flutter.dev/docs/cookbook/networking/background-parsing
and put json parsing into `compute`, moving it to the background.

Reply:
Same for other places that parse json